### PR TITLE
Allow only positive transfer values

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Allow only positive transfer values

Related issues:
- https://github.com/ainblockchain/ain-blockchain/issues/1208

### CODE REVIEW

File: src/wallet.ts [hunk 1]
```diff
@@ -256,6 +256,9 @@ export default class Wallet {
   transfer(input: {to: string, value: number, from?: string, nonce?: number, gas_price?: number}, isDryrun: boolean = false): Promise<any> {
     const address = this.getImpliedAddress(input.from);
     const toAddress = Ain.utils.toChecksumAddress(input.to);
+    if (!(input.value > 0)) {
```
Comment (woomurf) [inline src/wallet.ts:259]: Is there any reason to check value is positive number? I know, If value is 0 or negative number, blockchain will throw error.

**Review by woomurf — APPROVED** [review]
LGTM with some question. Thanks!

File: src/wallet.ts (hunk 1 above)
Comment (platfowner) [reply src/wallet.ts:259]: Right, this saves user's tx fee though.

**Comment by platfowner** [comment]
Thanks for the review! @woomurf  Now merging..

File: src/wallet.ts (hunk 1 above)
Comment (woomurf) [reply src/wallet.ts:259]: Oh, if the transaction fails, does the user still have to pay the transaction fee?

Comment (platfowner) [reply src/wallet.ts:259]: That's a good quesiton.
It depends on the conditions the transaction breaks.
If the're prechecks or skipFees = true, tx fee is not charged (see https://github.com/ainblockchain/ain-blockchain/blob/master/db/index.js#L1955).

Comment (woomurf) [reply src/wallet.ts:259]: Thank you for the detailed explanation!

**Pull request merged**